### PR TITLE
Reverse String in Kotlin

### PR DESCRIPTION
**Pull request by harshcrop**
Congrats on taking the first step to contributing to the [Sample Programs in Every Language][1] repository!

For simplicity, please make sure that your pull request includes one and only one sample program and language.

## Please Complete the Following

- [x] I would **NOT** like credit for my contribution in the article (credit will be included otherwise)
- [x] I plan to write the article for this code snippet
- [x] I named the pull request using `<Sample Program> in <Language>` format
- [x] I created/updated the language README
  - [x] I added the sample program name to the README
  - [x] I added fun facts (i.e. debut, developer, typing, etc.)
  - [x] I added reference link(s) to the README
  - [x] I added solution citations when necessary (see [plagiarism][2])
- [x] I fixed #your-issue-number-here

## Notes

Feel free to put whatever you want here.

[1]: https://therenegadecoder.com/code/sample-programs-in-every-language/
[2]: https://github.com/TheRenegadeCoder/sample-programs/blob/master/CONTRIBUTING.md#plagiarism

### CODE REVIEW

File: archive/k/kotlin/reverse-string.kt [hunk 1]
```diff
@@ -0,0 +1,12 @@
+fun String.reverseCaseOfString(): String {
+    val inputCharArr = toCharArray() // Converting the input to char array
+    var output = ""
+    for (i in 0 until inputCharArr.size) {
+        output += if (inputCharArr[i].isUpperCase()) { // Checking the character is in uppercase or not
```
[the remark this answers is not in the document]
Comment (jrg94) [reply archive/k/kotlin/reverse-string.kt]: Yeah, I'm thinking there's some confusing here. The method looks like it is "reversing the case" of a string--not reversing the indices.

File: archive/k/kotlin/reverse-string.kt [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+            inputCharArr[i].toLowerCase() // Converting the char to lower case
+        } else {
+            inputCharArr[i].toUpperCase() // Converting the char to upper case
+        }
+    }
+    return output
+}
```
Comment (jrg94) [inline archive/k/kotlin/reverse-string.kt]: Also, this ought to execute on its own. I'm not familiar with Kotlin, but if it's related to Java at all then you can make a call to this method in the main method. Just make sure to pull the string to be reversed off the command line.

**Comment by chris-c-thomas** [comment]
There is a reverse() function in Kotlin that can be utilized too.

`
fun main(args: Array<String>) {
 val reverse = args[0].reversed()
 println(reverse);
}
`

**Comment by jrg94** [comment]
Hey @harshcrop! @Barryrowe shared the proper solution in #571, so I'm going to close this solution If you'd like to propose your solution as a new sample program, I recommend editing the contributing doc with a description of what your program does (flip case vs. reverse string).

**Pull request closed without merging**